### DISCUSSION TITLE
Configuration error handling in pytest plugins

### DISCUSSION THREAD

**Discussion by bwoodsend** [pytest-dev/pytest#10211] [Q&A] [answered]
In the event that a pytest plugin realises that it has been misconfigured, is there a defacto way in which it should report that information to the user? For example,

```python
def pytest_collection_modifyitems(session, config, items):
    some_setting = config.getini("some-setting")
    if some_setting doesn't make sense:
        # What should go here to either abort the test run or propagate into a meaningful warning?
```
My initial instinct was just to raise an exception but doing so leads to a stacktrace being dumped with the prefix `INTERNALERROR>` all over it, then tests continuing as usual which incorrectly hints that something is broken deep in the bowels of pytest rather than in the user's setup.

**Comment by nicoddemus** [comment, marked as answer]
Try raising `pytest.UsageError`, we catch this at the top level and produce a shorter message.

**Reply by bwoodsend** [reply]
Oh, I do like the way pytest seems to think of everything :slightly_smiling_face:
